enhancements/installer/kubevirt-ipi.md

```diff
@@ -0,0 +1,342 @@
+
+`KubeVirt` is a virtualization platform running as an extension of Kubernetes.
+
+ We want to create a tenant cluster on top of existing OpenShift/Kubernetes cluster by creating
+ virtual machines by KubeVirt for every node in the tenant cluster (master and workers nodes)
+ and other Openshift/Kubernetes resources to allow **users** (not admins) of the infra cluster
+ to create a tenant cluster as it was an application running on the infra cluster.
+ We will implement all the components needed for the installer and cluster-api-provider
+ for the machine-api to allow day 2 operations of resizing the cluster.
+
+
+## Motivation
+
+- Achieve true multi-tenancy of OpenShift were each tenant has dedicated control plane 
+ and has full control on its configuration. 
+
+### Goals
+
+- provide a way to install OpenShift on KubeVirt infrastructure using 
+  the installer - an IPI installation. (1st day)
+- implementing a cluster-api provider to provide scaling and managing the cluster
+  nodes (used by IPI, and useful for UPI, and also node management/fencing) (2nd day)
+- Provide multi-tenancy and isolation between the tenant clusters
+
+### Non-Goals
+- UPI implementation will be provided separately.
+
+## Proposal
+
+This provider enables the OpenShift Installer to provision VM resources in 
+KubeVirt infrastructure, that will be used as worker and masters of the clusters. It 
+will also create the bootstrap machine, and the configuration needed to get
+the initial cluster running by supplying a DNS service and load balancing.
+
+We want to approach deployment on OpenShift and KubeVirt as deployment on cloud similar to the
+deployments we have on public clouds as AWS and GCP rather than virtualization platform in a way 
+that the machine's network will be private, and the relevant endpoints will be exposed out of the
+cluster with platform services as we can or pods deployed in the infrastructure cluster to supply the services
+as DNS and Loadbalancing.
+
+We see two main network options for deployment over KubeVirt:
+- Deploy the tenant cluster on the pods network and use OpenShift services and routes
+to provide DNS and Load-Balancing.
+- Deploy the tenant cluster on a secondary network (using Multus) and provide DNS service and Load-Balancing
+as the same way as other KNI networking deployments using HAProxy, CoreDNS and keepalived running on the
+tenant cluster VMs. See the [baremetal ipi networking doc][baremetal-ipi-networking]
+
+ 
+
+### Implementation Details/Notes/Constraints [optional]
+
+1. Survey
+
+    The installation starts and right after the user supplies their public ssh key,
+and then choose `KubeVirt` the installation will ask for all the relevant details
+of the installation: **kubeconfig** for the infrastructure OpenShift, **namespace**, **storageClass**, 
+ **networkName (NAD)** and other KubeVirt specific attributes. 
+The installer will validate it can communicate with the api, otherwise it will fail to proceed.
```

Comment:
have you thought through the negative flow and remediation when a failure occurs?  ideally the users knows exactly why this failed and how to diagnose, address and fix.  And we need to retain any input values the user chose or entered leading up to this point so he/she does not have to enter them again.  Same concept should apply for all input values we require.

Reply:
@robyoungky I don't see how this should be different from other platforms. At the end the user will have the same user experience as with any other IPIs. (while I agree this information can probably be  improved, it is out of  this enhancement scope)

Reply:
The installer config file from any of the IPI installers is persisted, and can be used to deploy the cluster again.